keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ArtifactController.kt

```diff
@@ -151,6 +157,9 @@ class ArtifactController(
     @PathVariable name: String,
     @PathVariable type: ArtifactType
   ): List<String> =
+    // FIXME: we can't authorize this call at the moment because neither the application name nor the delivery config
+    //  name are passed into this method. This means this method would also currently return artifact versions across
+    //  applications if the artifact names and types match, which I think is a bug.
```

Comment:
I don't think we need permissions on this call. Artifact versions are stored independently of applications, which is intentional and needed to prevent a potentially huge amount of data duplication. The available versions of an artifact is not restricted data.

Reply:
OK, I'll remove the `FIXME`. What about the question on returning versions across applications? Isn't the UI always going to have to index things based on application anyway? Seems like a waste of bandwidth to return the artifact versions for all apps if the artifact names happen to match (I have no idea how common that is).

Reply:
What do you mean? This is an api that pulls the artifact versions for a specific name and type of artifact. Artifacts are not scoped to an application.

Reply:
Oh, that's right, I totally forgot about that! I'm even using one of your container images in one of my delivery configs. 😄 I guess my confusion came from the fact that our top-level concept is the delivery config, and _it_ is scoped to an application. But the artifacts defined within it are not.